### PR TITLE
Specify ver

### PR DESCRIPTION
**Pull request by mirpo**

### CODE REVIEW

File: src/check-versions.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const check = require('check-node-version')
+const { engines } = require('../package.json')
+
+check(
```
Comment (harbu) [inline src/check-versions.js]: Should we put this in folder `$PROJECT_ROOT/bin`? It is a pattern I have seen in node projects to put utilities/tools in a folder called bin.

File: package.json [hunk 2]
```diff
@@ -35,6 +37,7 @@
   "engineStrict": true,
   "dependencies": {
     "async": "2.6.1",
+    "check-node-version": "^3.2.0",
```
Comment (kare) [inline package.json]: Please specify exact versions only from now on!

You can do either:
    % npm config set save-exact true
or
    % npm install --save --save-exact [package]

File: README.md [hunk 3]
```diff
@@ -1,10 +1,36 @@
...
+#### install current node lts (v8.12.0)
+
+`nvm install v8.12.0`
+
+`nvm use default v8.12.0`
+
+#### install npm (v6.4.1)
+
+`npm install -g npm@6.4.1`
+
```
Comment (kare) [inline README.md:29]: I'm thinking there is a chicken or egg problem here.

npm i -g check-node-version didn't help.

```
~/src/github.com/streamr-dev/network specify_ver  % npm --version
6.4.1
~/src/github.com/streamr-dev/network specify_ver  % node --version
v8.12.0
~/src/github.com/streamr-dev/network specify_ver  % npm i

> @streamr/streamr-p2p-network@1.0.3 preinstall /Users/kkn/src/github.com/streamr-dev/network
> npm run requirements-check


> @streamr/streamr-p2p-network@1.0.3 requirements-check /Users/kkn/src/github.com/streamr-dev/network
> node src/check-versions.js

module.js:550
    throw err;
    ^

Error: Cannot find module 'check-node-version'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/kkn/src/github.com/streamr-dev/network/src/check-versions.js:1:77)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @streamr/streamr-p2p-network@1.0.3 requirements-check: `node src/check-versions.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @streamr/streamr-p2p-network@1.0.3 requirements-check script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kkn/.npm/_logs/2018-09-19T09_11_11_920Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @streamr/streamr-p2p-network@1.0.3 preinstall: `npm run requirements-check`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @streamr/streamr-p2p-network@1.0.3 preinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kkn/.npm/_logs/2018-09-19T09_11_11_952Z-debug.log
```

Comment (kare) [reply README.md:29]: Perhaps we can work around this with something like:
Update README with "npm install -g check-node-version"

scripts: {
    preinstall: "check-node-version --node '>= 8.12.0' && ...",
}

**Review by kare — CHANGES_REQUESTED** [review]
Check comments!

File: package.json (hunk 2 above)
Comment (harbu) [reply package.json]: Fixed. Btw should this be a devDependency instead of dependency?

**Comment by harbu** [comment]
- I put exact version of the `check-node-version` in package.json file.
- I took the liberty of moving the script to the bin folder since no one objected. 😶

File: README.md (hunk 3 above)
Comment (harbu) [reply README.md:29]: You are saying that even after installing the module globally on your system, the `require` will not work?

Comment (kare) [reply README.md:29]: @harbu Yeah that's what happens. Just tried it again

I've found this: https://github.com/npm/npm/issues/20434 I think this is the issue we have experienced when running "npm install"  and observing changes in package-lock.json even though package.json is unmodified. It seems to be a non issue. Also learned about "npm ci".

**Pull request merged**